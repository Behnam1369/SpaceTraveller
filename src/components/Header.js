import React from 'react';
import { NavLink } from 'react-router-dom';
import * as url from '../assets/planet.png';

const Header = () => (
  <header>
    <NavLink to="/">
      <img className="logo" src={url.default} alt="space-logo" />
    </NavLink>
    <h1>Space Travelers&apos; Hub</h1>
    <nav>
      <NavLink to="/">Rockets</NavLink>
      <NavLink to="/Missions">Missions</NavLink>
      <NavLink to="/MyProfile">My Profile</NavLink>
    </nav>
  </header>
);

export default Header;
