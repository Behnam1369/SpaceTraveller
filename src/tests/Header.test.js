import TestRenderer from 'react-test-renderer';
import { BrowserRouter as Router } from 'react-router-dom';
import Header from '../components/Header';
import '@testing-library/jest-dom';

describe('Testing Header', () => {
  it('Testing Header component snapshot', () => {
    const testRenderer = TestRenderer.create(
      <Router>
        <Header />
      </Router>,
    ).toJSON();
    expect(testRenderer).toMatchSnapshot();
  });
});
