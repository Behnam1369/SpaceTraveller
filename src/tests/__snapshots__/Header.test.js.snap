// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing Header Testing Header component snapshot 1`] = `
<header>
  <a
    aria-current="page"
    className="active"
    href="/"
    onClick={[Function]}
  >
    <img
      alt="space-logo"
      className="logo"
      src="planet.png"
    />
  </a>
  <h1>
    Space Travelers' Hub
  </h1>
  <nav>
    <a
      aria-current="page"
      className="active"
      href="/"
      onClick={[Function]}
    >
      Rockets
    </a>
    <a
      className=""
      href="/Missions"
      onClick={[Function]}
    >
      Missions
    </a>
    <a
      className=""
      href="/MyProfile"
      onClick={[Function]}
    >
      My Profile
    </a>
  </nav>
</header>
`;
